src/scvi/dataloaders/_data_splitting.py

```diff
@@ -118,24 +181,38 @@ def __init__(
         self.load_sparse_tensor = load_sparse_tensor
         self.data_loader_kwargs = kwargs
         self.pin_memory = pin_memory
+        self.use_external_indexing = use_external_indexing
+        self.external_indexing = external_indexing
 
         self.n_train, self.n_val = validate_data_split(
-            self.adata_manager.adata.n_obs, self.train_size, self.validation_size
+            self.adata_manager.adata.n_obs,
+            self.train_size,
+            self.validation_size,
+            self.use_external_indexing,
+            self.external_indexing,
```

Comment:
Since `validate_data_split` is now trying to handle two sets of arguments (one for `train_size` and the other for `external_indexing`), maybe we can refactor this out into two functions instead. What do you think?

Reply:
Can you elaborate? Wouldn't this be much more challenging to implement and manage?

Reply:
change to sets in the beginning of validation and use that further on code.
divide only the validation part to 2 cases
We want to implement it in the semi-supervised class as well (divide to labeld/unlabled)